src/main/java/htsjdk/samtools/util/AbstractRecordAndOffset.java

```diff
@@ -59,14 +53,16 @@
      * @param refPos corresponding to read offset reference position
      */
     public AbstractRecordAndOffset(final SAMRecord record, final int offset, int length, int refPos) {
-        validateIndex(offset, record.getBaseQualities());
-        if (length > record.getReadLength()) {
-            throw new IllegalArgumentException("Block length cannot be larger than whole read length");
-        }
+        this(record, offset);
+    }
+
+    /**
+     * @param record inner SAMRecord
+     * @param offset from the start of the read
```

Comment:
the code and comments are somewhat contradictory: what is offset? here it says "offset from the strart of read" but in getBaseQuality it talks about the reference....